src/main/java/htsjdk/samtools/SAMSequenceRecord.java

```diff
@@ -98,68 +107,185 @@ public SAMSequenceRecord(final String name, final int sequenceLength) {
         mSequenceLength = sequenceLength;
     }
 
-    public String getSequenceName() { return mSequenceName; }
+    public String getSequenceName() {
+        return mSequenceName;
+    }
 
-    public int getSequenceLength() { return mSequenceLength; }
-    public void setSequenceLength(final int value) { mSequenceLength = value; }
+    public int getSequenceLength() {
+        return mSequenceLength;
+    }
 
-    public String getAssembly() { return (String) getAttribute(ASSEMBLY_TAG); }
-    public void setAssembly(final String value) { setAttribute(ASSEMBLY_TAG, value); }
+    public void setSequenceLength(final int value) {
+        mSequenceLength = value;
+    }
 
-    public String getSpecies() { return (String) getAttribute(SPECIES_TAG); }
-    public void setSpecies(final String value) { setAttribute(SPECIES_TAG, value); }
+    public String getAssembly() {
+        return (String) getAttribute(ASSEMBLY_TAG);
+    }
 
-    public String getMd5() { return (String) getAttribute(MD5_TAG); }
-    public void setMd5(final String value) { setAttribute(MD5_TAG, value); }
+    public void setAssembly(final String value) {
+        setAttribute(ASSEMBLY_TAG, value);
+    }
 
-    public String getDescription() { return getAttribute(DESCRIPTION_TAG);}
-    public void setDescription(final String value) { setAttribute(DESCRIPTION_TAG, value);}
+    public String getSpecies() {
+        return (String) getAttribute(SPECIES_TAG);
+    }
+
+    public void setSpecies(final String value) {
+        setAttribute(SPECIES_TAG, value);
+    }
+
+    public String getMd5() {
+        return (String) getAttribute(MD5_TAG);
+    }
+
+    public void setMd5(final String value) {
+        setAttribute(MD5_TAG, value);
+    }
+
+    public String getDescription() {
+        return getAttribute(DESCRIPTION_TAG);
+    }
+
+    public void setDescription(final String value) {
+        setAttribute(DESCRIPTION_TAG, value);
+    }
 
     /**
-     * @return Index of this record in the sequence dictionary it lives in. 
+     * @return Index of this record in the sequence dictionary it lives in.
      */
-    public int getSequenceIndex() { return mSequenceIndex; }
+    public int getSequenceIndex() {
+        return mSequenceIndex;
+    }
 
     // Private state used only by SAM implementation.
-    public void setSequenceIndex(final int value) { mSequenceIndex = value; }
+    public void setSequenceIndex(final int value) {
+        mSequenceIndex = value;
+    }
+
+    /**
+     * Returns unmodifiable set with alternative sequence names.
+     */
+    public Set<String> getAlternativeSequenceNames() {
+        final String anTag = getAttribute(ALTERNATIVE_SEQUENCE_NAME_TAG);
+        return (anTag == null) ? Collections.emptySet()
+                : Collections.unmodifiableSet(new HashSet<>(Arrays.asList(anTag.split(ALTERNATIVE_SEQUENCE_NAME_SEPARATOR))));
```

Comment:
This shouldn't be coerced through a HashSet, it will shuffle the order.  Use a LinkedHashSet instead.